### PR TITLE
HLSL: fix preprocessor concatenation behaviour.

### PR DESCRIPTION
**Pull request by xxxbxxx**

### CODE REVIEW

**Comment by johnkslang** [comment]
This PR has a problem, beyond failing tests, of replicating another PR, rather than being independent of it.

It would be good to see the just the preprocessor fix as its own PR.

**Comment by xxxbxxx** [comment]
updated.

**Comment by johnkslang** [comment]
This issue came up in #772. Does this get the correct results for the test there?  Or, you prefer I verify?

File: glslang/MachineIndependent/preprocessor/Pp.cpp [hunk 1]
```diff
@@ -1068,6 +1068,10 @@ int TPpContext::tMacroInput::scan(TPpToken* ppToken)
         pasting = true;
     }
 
+    // HLSL fxc.exe does expand macros before concatenation
```
Comment (johnkslang) [inline glslang/MachineIndependent/preprocessor/Pp.cpp]: I hope this is part of HLSL, no matter the front end, not an aspect of fxc.  Can we remove the "fxc.exe"?

**Comment by xxxbxxx** [comment]
I added #772 as a test case and compared the output with fxc and it looks the same.

(I have no idea where the hlsl preprocessor behaviour is documented.)

**Comment by johnkslang** [comment]
Looks good, thanks.

**Pull request merged**